src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -223,6 +224,20 @@ private bool MatchString(string str)
             {
                 if (runtextend - runtextpos < c)
                 {
+                    // If MatchString was called after a greedy op such as a .*, we would have zipped runtextpos to the end without really examining any characters. Reset to maxBacktrackPos here as an optimization
+                    if (!_caseInsensitive && _maxBacktrackPosition != -1 && runtextpos > _maxBacktrackPosition)
+                    {
+                        // If lastIndexOf is -1, we backtrack to the max extent possible.
+                        runtextpos = _maxBacktrackPosition;
+                        ReadOnlySpan<char> runtextSpan = runtext.AsSpan(_maxBacktrackPosition);
```

Comment:
Should this be bounded by runtextend?

Reply:
Ah yes, especially after seeing your PR from yesterday. Fixed now.